scripts/analyze.py

```diff
@@ -13,10 +14,28 @@ def get_project_fuzzer_stats(project: str) -> pd.DataFrame:
     proj = Project(project, fuzzout, dumpout)
     return proj.get_project_stats()
 
+def get_proj_name(line: str) -> Optional[str]:
+    #pattern = '^[0-9]+\. \[?([A-Za-z0-9\-\_]+)\]?\([A-Za-z\.\_\-/ +]+\),([A-Za-z+]+)'
+    # Group 1 is project name, group 2 is project language
+    pattern = '^[0-9]+\. \[?([^\]]+)\]?\(.+\),([A-Za-z+]+)'
+    
+    line = line.strip()  # remove space at front and end
+    match = re.match(pattern, line)
+
+    if not match:
+        return None
+    
+    proj_name = match.group(1)
+    proj_language = match.group(2)
+    return proj_name
 
 def main():
-    projects = ["clib", "coturn", "croaring", "gpac", "h3", "inchi", "libdwarf", "libpg_query", "libssh", "libyang", "lua",
-            "llhttp", "md4c", "miniz", "tmux", "p11-kit", "pycryptodome", "quickjs", "utf8proc", "w3m", "zydis"]  # todo: full list of built & fuzzed projects
+    
+    with open("./built_w_pass.md", "r") as f:
+        lines = f.readlines()
+    projects = list(filter(lambda x: x is not None, map(get_proj_name, lines))) 
```

Comment:
here use `map(lambda l: get_proj_info(l)[0], lines)` or some list comprehension is also okay. `list()` is not necessary since we can just use the iterator later

Reply:
Should I use list comprehension to extract the project names into a list?

Reply:
Oh wait never mind I think I get what you mean

Reply:
How should we handle calling analyze for a specific programming language? Right now my modified analyze.py will get the list of all of the project names and then crash since we do not have the c++ projects ready to be analyzed yet.

Reply:
modify `get_proj_info` to return a tuple of str, i.e. `(proj_name, proj_lang)`

then do something like

```py
projects = [name for name, lang in map(get_proj_info, lines) if lang == 'c']
```